scripts/docker-publish-full.js

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/env node
+
+/**
+ * This should normally be run without any environment changes and will build, tag, and push 2 docker images for latest-rc
+ * Should you ever need to manually run this script, then
+ * 1. make sure you've logged into docker from its CLI `docker login`
+ * 3. provide the version, example: SALESFORCE_CLI_VERSION=7.100.0 ./scripts/docker-publish
+ * 4. you can add NO_PUBLISH=true if you want to only do local builds from the script
+ */
+const shell = require('shelljs');
+const fs = require('fs-extra');
+const dockerShared = require('./docker-shared');
+
+shell.set('-e');
+shell.set('+v');
+
+const DOCKER_HUB_REPOSITORY = dockerShared.repo;
+
+(async () => {
+  dockerShared.validateDockerEnv();
+  const SALESFORCE_CLI_VERSION = await dockerShared.getCliVersion();
+
+  shell.exec(
+    `docker build --file ./dockerfiles/Dockerfile_full --build-arg SALESFORCE_CLI_VERSION=${SALESFORCE_CLI_VERSION} --tag ${DOCKER_HUB_REPOSITORY}:${SALESFORCE_CLI_VERSION}-full --no-cache .`
+  );
+
+  if (process.env.NO_PUBLISH) return;
+
+  // Push to the Docker Hub Registry
+  shell.exec(`docker push ${DOCKER_HUB_REPOSITORY}:${SALESFORCE_CLI_VERSION}-full`);
+
+  // This normally defaults to latest-rc.  If you've supplied it in the environment, we're not tagging latest-rc.
+  if (process.env['SALESFORCE_CLI_VERSION']) return;
+
+  // tag the newly created version as latest-rc
+  shell.exec(
+    `docker tag ${DOCKER_HUB_REPOSITORY}:${SALESFORCE_CLI_VERSION}-full ${DOCKER_HUB_REPOSITORY}:latest-rc-full`
+  );
+  shell.exec(`docker push ${DOCKER_HUB_REPOSITORY}:latest-rc-full`);
```

Comment:
Can you remind me what `rc` stands for? Is it relevant info to the customer?

Is there a particular reason you want to re-name the published tags? I think this will "break" existing customers who have `FROM saleforce/salesforcedx:latest-full` in their CI, except they may not realize the version of the CLI is no longer the latest until they change it to `latest-rc-full`.

same with the slim image.

Reply:
`rc` is the release candidate that gets promote to latest the following week.  Here's a quick explanation of how it works for CLI: https://github.com/salesforcecli/sfdx-cli#releases

There still is a latest-full.  The `docker-promote.js` tags what was `latest-rc-full/slim` as `latest-full/slim`

But if people want to run on the release candidate, they can.  The other thing our team like about this is that it follows our release process (we build rc for tarballs, binary installers, npm tags) and then promote to latest the following week.

Reply:
> I am also curious why do you need to install the CLI in the slim and full differently?

slim is slimmer (only the bundled node version).  The main thing it does is give us coverage for both of our typical install paths (npm and binary).  

The docker builds are serving as "the rc release is usable in both ways" and makes it easy to do things like 1) repro an issue using a docker image when my machine uses the other install approach 2) use the images in CI

Reply:
ah, I missed line 22 in `docker-promote.js`, thanks. The rest all makes sense.